pip/pip-425.md

```diff
@@ -0,0 +1,127 @@
+# PIP-425: Support connecting with next available endpoint for multi-endpoint serviceUrls
+Implementation PR: [#24387](https://github.com/apache/pulsar/pull/24387)
+
+# Background knowledge
+In the current Pulsar client versions, consumer or producer creation is the responsibility of PulsarClient, we can build a PulsarClient by pass serviceUrl param. The serviceUrl param supports multiple endpoints with the syntax `pulsar://localhost1:6650,localhost2:6650,localhost3:6650,...` or `pulsar+ssl://localhost1:6650,localhost2:6650,localhost3:6650,...`. What happens if some of the endpoints are unavailable? Yes, in the current version, it may cause the consumer or producer to fail to create, which is also the optimization goal of this pip.
+
+```java
+public class Example {
+    public static void main(String[] args) throws PulsarClientException {
+        PulsarClient pulsarClient = PulsarClient.builder()
+                .serviceUrl("pulsar://localhost1:6650,localhost2:6650,localhost3:6650")
+                .build();
+        Consumer<Long> consumer = pulsarClient.newConsumer(Schema.INT64)
+                .topic("persistent://public/default/my-topic")
+                .subscriptionName("my-subscription")
+                .subscribe();
+
+        Producer<Long> producer = pulsarClient.newProducer(Schema.INT64)
+                .topic("persistent://public/default/my-topic")
+                .create();   
+
+        .....
+
+    }
+}
+
+```
+
+# Motivation
+
+As [#22934 ](https://github.com/apache/pulsar/issues/22934) and [#22933](https://github.com/apache/pulsar/discussions/22933) mentioned, when most of the endpoints in serviceurl are unavilable (but there is at least one available endpoint), creating consumers and producers through PulsarClient will most likely fail. I think this is not as expected. If the code is robust enough, as long as there is one available endpoint, it should be accessible normally. Obviously, we can remove unavailable endpoints through some mechanism, and try to ensure the next connection to an available endpoint. 
```

Comment:
It's better to avoid references to other sources in the PIP motivation part. You might combine the "background knowledge" and "motivation" sections into a single section.

In the writing style it's better to avoid personal style, instead of "I think this is not expected", you could just make the claim which some rationale, for example "this is unexpected behavior for Pulsar users since when multiple endpoints are provided, it would be expected that when an endpoint is unavailable, an available endpoint would be used", etc.